File: python/3622.py
class Solution(object):
    def checkDivisibility(self, n):
        """
        :type n: int
        :rtype: bool
        """
        digit = str(n)
        addition = 0
        multiplication = 1
        for i in digit:
            addition = addition + int(i)
            multiplication = multiplication * int(i)

        total = addition + multiplication
        if(total==n):
            return True
        else:
            if n % total == 0:
                return True
            return False
            

s = Solution()
n = int(input("Enter number you want to enter it: "))
print(s.checkDivisibility(n))

File: Javascript/3622.js
/**
 * @param {number} n
 * @return {boolean}
 */
var checkDivisibility = function(n) {
    digit = String(n);
    let addition = 0;
    let multiplication = 1;
    // console.log(digit, typeof(digit));
    // console.log(digit.length);
    for(i=0; i<digit.length;i++){
        // console.log(parseInt(i, 10), digit[i]) // not working
        addition = addition + Number(digit[i]);
        // console.log(addition);
        multiplication = multiplication * Number(digit[i]);
        // console.log(multiplication);
    }
    const total = addition + multiplication;
    // console.log(total, addition, multiplication);
    if(total == n){
        return true;
    } else if(total % n == 0){
        return true;
    } else{
        return false;
    }
};

const n = 99;
// console.log("Test case 1", checkDivisibility(n));
console.log(checkDivisibility(n));